core/src/main/java/org/apache/calcite/avatica/AvaticaConnection.java

```diff
@@ -276,14 +298,17 @@ public CallableStatement prepareCall(
   }
 
   public Map<String, Class<?>> getTypeMap() throws SQLException {
-    throw HELPER.unsupported();
+    checkOpen();
+    // Always return an empty map since custom types is not supported
+    return new HashMap<>();
```

Comment:
Would it be more correct to return `SQLFeatureNotSupportedException`? I don't think it matters either way but below is from the javadoc for Connection.

> SQLFeatureNotSupportedException - if the JDBC driver does not support this method

Reply:
yes, probably more correct. I think I got the truncated version of the javadoc on this one. I will revert.